pkg/codegen/internal/test/testdata/simple-resource-schema/otherResource.go

```diff
@@ -0,0 +1,68 @@
+// *** WARNING: this file was generated by test. ***
+// *** Do not edit by hand unless you're certain you know what you are doing! ***
+
+package example
+
+import (
+	"reflect"
+
+	"github.com/pulumi/pulumi/sdk/v2/go/pulumi"
+)
+
+type OtherResource struct {
+	pulumi.CustomResourceState
+
+	Foo pulumi.ResourceOutput `pulumi:"foo"`
+}
+
+// NewOtherResource registers a new resource with the given unique name, arguments, and options.
+func NewOtherResource(ctx *pulumi.Context,
+	name string, args *OtherResourceArgs, opts ...pulumi.ResourceOption) (*OtherResource, error) {
+	if args == nil {
+		args = &OtherResourceArgs{}
+	}
+	var resource OtherResource
+	err := ctx.RegisterResource("example::OtherResource", name, args, &resource, opts...)
```

Comment:
Just a note: In the draft PR of the SDK changes, I've added `RegisterRemoteComponentResource` that should be used for remote components (unless we want to expose this in some other way).

https://github.com/pulumi/pulumi/pull/5558/files#diff-6ac1847413cb55c8d480423da6e23ed6ac98c0066fcf7c57f8b5cd960a04dee1R526